packages/headless/src/features/search/search-actions.ts

```diff
@@ -235,7 +235,10 @@ const getStateAfterResponse: (
     ...getSearchInitialState(),
     duration,
     response,
-    results: response.results,
+    results: response.results.map((result) => ({
+      ...result,
+      searchUid: response.searchUid,
```

Comment:
This mapping would also need to be performed on the [results returned by the recommendation request](https://github.com/coveo/ui-kit/blob/master/packages/headless/src/features/recommendation/recommendation-actions.ts#L82).

We'll need to go through the code base and update type at certain responses to `SearchApiResult` so that typescript can help us,

Reply:
What if instead we do it inside the search-api-client and add a test